llvm/lib/Transforms/Vectorize/LoadStoreVectorizer.cpp

```diff
@@ -1625,8 +1782,7 @@ std::vector<Chain> Vectorizer::gatherChains(ArrayRef<Instruction *> Instrs) {
   Ret.reserve(Chains.size());
   // Iterate over MRU rather than Chains so the order is deterministic.
   for (auto &E : MRU)
-    if (E.second.size() > 1)
-      Ret.emplace_back(std::move(E.second));
+    Ret.emplace_back(std::move(E.second));
```

Comment:
Can you explain this change? Are you basically allowing chains that have a size of 1 in order to merge them later?